internal/config/config_test.go

```diff
@@ -210,8 +224,54 @@ func TestInvalidConfigurationValues(t *testing.T) {
 			err := updateConfig(cfg)
 			assert.Error(t, err, "invalid input should be rejected by validation")
 			assert.Contains(t, err.Error(), tc.expectedError)
+		})
+	}
+}
+
+func TestConfigString(t *testing.T) {
+	testCases := []struct {
+		name   string
+		config *Config
+	}{{
+		name: "default config",
+		config: &Config{
+			Log: Log{
+				Level:  "info",
+				Format: "text",
+			},
+		},
+	}, {
+		name: "custom config",
+		config: &Config{
+			Log: Log{
+				Level:  "debug",
+				Format: "json",
+			},
+		},
+	}}
+
+	// test yaml marshall
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			// Get string representation
+			str := tc.config.String()
 
+			// Verify it's valid YAML and contains the expected values
+			assert.Contains(t, str, "log:")
+			assert.Contains(t, str, "level: "+tc.config.Log.Level)
+			assert.Contains(t, str, "format: "+tc.config.Log.Format)
 		})
 	}
 
+	// test manual string builder approach
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			// Get string representation
+			str := tc.config.manualString()
+
+			// Verify it's valid YAML and contains the expected values
+			assert.Contains(t, str, "log.level: "+tc.config.Log.Level)
+			assert.Contains(t, str, "log.format: "+tc.config.Log.Format)
```

Comment:
question: is it a deliberate choice to have "level:" in case of yaml marshalling and "log.level" in case of manual ?

Reply:
lgtm otherwise

Reply:
yes, that was intentional. It tells us the config failed to Marshall properly (for whatever reason).